js/react_native/e2e/ios/Podfile

```diff
@@ -1,28 +1,76 @@
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, '12.4'
+platform :ios, min_ios_version_supported
+prepare_react_native_project!
+
+# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
+# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
+#
+# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
+# ```js
+# module.exports = {
+#   dependencies: {
+#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
+# ```
+flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
+
+linkage = ENV['USE_FRAMEWORKS']
+if linkage != nil
+  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
+  use_frameworks! :linkage => linkage.to_sym
+end
+
+# Use New Architecture
+# ENV['RCT_NEW_ARCH_ENABLED'] = '1'
 
 target 'OnnxruntimeModuleExample' do
   config = use_native_modules!
 
-  use_react_native!(:path => config["reactNativePath"])
+  # Flags change depending on the env values.
+  flags = get_default_flags()
 
-  use_frameworks!
+  use_react_native!(
+    :path => config[:reactNativePath],
+    # Hermes is now enabled by default. Disable by setting this flag to false.
+    # Upcoming versions of React Native may rely on get_default_flags(), but
+    # we make it explicit here to aid in the React Native upgrade process.
+    :hermes_enabled => flags[:hermes_enabled],
+    :fabric_enabled => flags[:fabric_enabled],
+    # Enables Flipper.
+    #
+    # Note that if you have use_frameworks! enabled, Flipper will not work and
+    # you should disable the next line.
+    :flipper_configuration => flipper_config,
+    # An absolute path to your application root.
+    :app_path => "#{Pod::Config.instance.installation_root}/.."
+  )
 
   # Mobile build:
   # ort_mobile_c_local_pod_path = ENV['ORT_MOBILE_C_LOCAL_POD_PATH']
   # if ort_mobile_c_local_pod_path != nil
   #   print 'Using onnxruntime-c pod at ', ort_mobile_c_local_pod_path, "\n"
   #   pod 'onnxruntime-mobile-c', :path => ort_mobile_c_local_pod_path
   # end
-  
   ort_c_local_pod_path = ENV['ORT_C_LOCAL_POD_PATH']
   if ort_c_local_pod_path != nil
     print 'Using onnxruntime-c pod at ', ort_c_local_pod_path, "\n"
     pod 'onnxruntime-c', :path => ort_c_local_pod_path
   end
   pod 'onnxruntime-react-native', :path => '../node_modules/onnxruntime-react-native'
 
-  inherit! :search_paths
+  target 'OnnxruntimeModuleExampleTests' do
+    inherit! :complete
+    # Pods for testing
+  end
+
+  post_install do |installer|
+    react_native_post_install(
+      installer,
+      # Set `mac_catalyst_enabled` to `true` in order to apply patches
+      # necessary for Mac Catalyst builds
+      :mac_catalyst_enabled => false
+    )
+    __apply_Xcode_12_5_M1_post_install_workaround(installer)
```

Comment:
nit: is this a specific workaround for your dev version? (Xcode 12.5 & M1)?  afaik, we are using xcodeversion 14.3 in the CI and no M1 support yet for the testing machine agent.

Reply:
It was taken from the official RN template, and I don't think it's needed anymore so it can be removed.